### PR TITLE
Pods 8198

### PR DESCRIPTION
**Pull request by ngmjohnstone**

### CODE REVIEW

File: app/transformations/ETMPToFrontEnd/MemberEventReport.scala [hunk 1]
```diff
@@ -16,31 +16,265 @@
...
+    case Event4 =>          rdsMemberDetailsEvent4
+    case Event5 =>          rdsMemberDetailsEvent5
+    case Event6 =>          rdsMemberDetailsEvent6
+    case Event7 =>          rdsMemberDetailsEvent7
+    case Event8 =>          rdsMemberDetailsEvent8
+    case Event8A =>         rdsMemberDetailsEvent8A
+    case _ =>               rdsMemberDetailsEvent22And23
+  }
+
+  implicit val rdsMemberDetailsEvent2:        Reads[JsObject] = (readsDeceasedMemberDetails and readsBeneficiaryDetails and readsEvent2PaymentDetails).reduce
```
Comment (grantrobson) [inline app/transformations/ETMPToFrontEnd/MemberEventReport.scala:49]: I don't think these need to be implicit. They should be private too I think

File: test/utils/GeneratorAPI1832.scala [hunk 2]
```diff
@@ -17,55 +17,453 @@
...
+              "amountPaid" -> map("pensionAmt")
+            )
+          )
+        )
+      )
+      Tuple2(etmpPayload, userAnswers)
+    }
+  }
+
+  def generateUAFromETMPDataForEvent3: Gen[(JsObject, JsObject)] = {
```
Comment (grantrobson) [inline test/utils/GeneratorAPI1832.scala:100]: Make private

File: test/utils/GeneratorAPI1832.scala [hunk 3]
```diff
@@ -17,55 +17,453 @@
...
+              ),
+            )
+          )
+        )
+      )
+      Tuple2(etmpPayload, userAnswers)
+    }
+  }
+
+  def generateUAFromETMPDataForEvent4: Gen[(JsObject, JsObject)] = {
```
Comment (grantrobson) [inline test/utils/GeneratorAPI1832.scala:151]: Make private

File: test/utils/GeneratorAPI1832.scala [hunk 4]
```diff
@@ -110,11 +508,113 @@ trait GeneratorAPI1832 extends Matchers with OptionValues with ResponseGenerator
...
     for {
       chosenEventType <- Gen.oneOf[EventType](Seq(EventType.Event22, EventType.Event23))
     } yield {
       Tuple2(generatedPayload(chosenEventType), chosenEventType)
     }
   }
 }
+
+
+//noinspection ScalaStyle
```
Comment (grantrobson) [inline test/utils/GeneratorAPI1832.scala:521]: Why we switching off scalastyle here?

File: app/transformations/ETMPToFrontEnd/MemberEventReport.scala (hunk 1 above)
Comment (ngmjohnstone) [reply app/transformations/ETMPToFrontEnd/MemberEventReport.scala]: Addressed here: https://github.com/hmrc/pension-scheme-event-reporting/pull/74/commits/7192c98d8dcdc93ebebf3355c39e0b712a2a81f6

File: test/utils/GeneratorAPI1832.scala (hunk 2 above)
Comment (ngmjohnstone) [reply test/utils/GeneratorAPI1832.scala]: Addressed here: https://github.com/hmrc/pension-scheme-event-reporting/pull/74/commits/7192c98d8dcdc93ebebf3355c39e0b712a2a81f6

File: test/utils/GeneratorAPI1832.scala (hunk 3 above)
Comment (ngmjohnstone) [reply test/utils/GeneratorAPI1832.scala]: Addressed here: https://github.com/hmrc/pension-scheme-event-reporting/pull/74/commits/7192c98d8dcdc93ebebf3355c39e0b712a2a81f6

File: test/utils/GeneratorAPI1832.scala (hunk 4 above)
Comment (ngmjohnstone) [reply test/utils/GeneratorAPI1832.scala]: Addressed here: https://github.com/hmrc/pension-scheme-event-reporting/pull/74/commits/7192c98d8dcdc93ebebf3355c39e0b712a2a81f6

**Review by grantrobson — APPROVED** [review]

**Pull request merged**